corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py

```diff
@@ -0,0 +1,178 @@
+        index_tree = IndexTree(indices=all_indices)
+        case_ids_on_phone = all_syncing
+        primary_cases_syncing = all_syncing - all_dependencies_syncing
+        if not self.restore_state.is_initial:
+            case_ids_on_phone = case_ids_on_phone | self.restore_state.last_sync_log.case_ids_on_phone
+            # subtract primary cases from dependencies since they must be newly primary
+            all_dependencies_syncing = all_dependencies_syncing | (
+                self.restore_state.last_sync_log.dependent_case_ids_on_phone -
+                primary_cases_syncing
+            )
+            index_tree = index_tree | self.restore_state.last_sync_log.index_tree
+
+        self.restore_state.current_sync_log.case_ids_on_phone = case_ids_on_phone
+        self.restore_state.current_sync_log.dependent_case_ids_on_phone = all_dependencies_syncing
+        self.restore_state.current_sync_log.index_tree = index_tree
+        # todo: update stock payload
+        return response
+
+    def get_case_ids_for_owner(self, owner_id):
+        if self.is_clean(owner_id):
+            if self.restore_state.is_initial:
+                # for a clean owner's initial sync the base set is just the open ids
+                return set(get_open_case_ids(owner_id))
+            else:
+                # for a clean owner's steady state sync, the base set is anything modified since last sync
+                return set(get_case_ids_modified_with_owner_since(
+                    self.restore_state.domain, owner_id, self.restore_state.last_sync_log.date
+                ))
+        else:
+            # todo: we may want to be smarter than this
+            # right now just return the whole footprint and do any filtering later
+            return get_case_footprint_info(self.restore_state.domain, owner_id).all_ids
+
+
+def _is_live(case, restore_state):
+    """
+    Given a case and a restore state object, return whether or not the case is "live"
+    (direclty owned by this sync and open), or "dependent" (needed by another case)
+    """
+    return not case.closed and get_owner_id(case) in restore_state.owner_ids
+
+
+def filter_cases_modified_since(domain, case_ids, reference_date):
+    """
+    Given a domain, case_ids, and a reference date, filter the case ids to only those
+    that have been modified since that reference date.
+    """
+    last_modified_date_dict = get_last_modified_dates(domain, case_ids)
+    for case_id in case_ids:
+        if last_modified_date_dict.get(case_id, datetime(1900, 1, 1)) > reference_date:
+            yield case_id
+
+
+def case_needs_to_sync(case, last_sync_log):
+    owner_id = case.owner_id or case.user_id  # need to fallback to user_id for v1 cases
+    if not last_sync_log or owner_id not in last_sync_log.owner_ids_on_phone:
+        # initial sync or new owner IDs always sync down everything
+        return True
+    elif case.server_modified_on < last_sync_log.date:
```

Comment:
Perhaps just preference but I'd swap these two round and return `True`. I just think it makes it easier to read since it keeps all the conditions positive (if this is true then sync)